flink-core/src/test/java/org/apache/flink/configuration/ConfigurationTest.java

```diff
@@ -435,6 +446,16 @@ public void testMapRemovePrefix() {
         assertFalse(cfg.containsKey(MAP_PROPERTY_2));
     }
 
+    @Test
+    public void testMapWorksWithNonStringValue() {
+        final Configuration cfg = new Configuration();
+        cfg.setString(MAP2_PROPERTY_1, "1");
+        cfg.setString(MAP2_PROPERTY_2, "2");
+
+        Map<String, Integer> stringIntegerMap = cfg.get(MAP_OPTION_INTEGER_TYPE);
+        assertEquals(1L, (long) stringIntegerMap.get("prop1"));
```

Comment:
Can you make this a long option to be bit less weird, without the casting to a long?